include/wx/event.h

```diff
@@ -1946,31 +1955,92 @@ class WXDLLIMPEXP_CORE wxSetCursorEvent : public wxEvent
 
 private:
     wxDECLARE_DYNAMIC_CLASS_NO_ASSIGN(wxSetCursorEvent);
+};
+
+ // Touch Base Event
+
+class WXDLLIMPEXP_CORE wxTouchEventBase : public wxEvent
+{
+public:
+    wxTouchEventBase(wxWindowID winid = 0, wxEventType type = wxEVT_NULL)
+        : wxEvent(winid, type)
+    {
+    }
+
+    wxTouchEventBase(const wxTouchEventBase& event) : wxEvent(event)
+    {
+        m_pos = event.m_pos;
+    }
+
+    const wxPoint& GetPosition() const { return m_pos; }
+    void SetPosition(const wxPoint& pos) { m_pos = pos; }
+
+protected:
+    wxPoint m_pos;
+
+    wxDECLARE_ABSTRACT_CLASS(wxTouchEventBase);
+
+};
+
+// MultiTouch Event
+
+class wxTouchSequenceId : public wxItemId<void*>
+{
+public:
+    wxTouchSequenceId() : wxItemId<void*>() { }
+    explicit wxTouchSequenceId(void* pItem) : wxItemId<void*>(pItem) { }
+};
+
+class WXDLLIMPEXP_CORE wxMultiTouchEvent : public wxTouchEventBase
+{
+public:
+    wxMultiTouchEvent(wxWindowID winid = 0, wxEventType type = wxEVT_NULL)
+        : wxTouchEventBase(winid, type)
+    {
+        m_isPrimary = false;
+    }
+
+    wxMultiTouchEvent(const wxMultiTouchEvent& event) : wxTouchEventBase(event)
+    {
+        m_isPrimary = event.m_isPrimary;
+        m_sequence = event.m_sequence;
+    }
```

Comment:
AFAICS this should be just
```suggestion
    wxMultiTouchEvent(const wxMultiTouchEvent& event) = default;
```